src/k8s-job-exit-spec/config/k8s-job-exit-spec.yaml

```diff
@@ -840,6 +848,202 @@ spec:
     runtimeContainerPatterns:
     - userLogRegex: "(?msi)CUDA_ERROR_ECC_UNCORRECTABLE.*"
 
+- code: 224
+  phrase: ContainerCudaVersionMismatch
+  issuer: PAI_RUNTIME
+  causer: USER_CONTAINER
+  type: USER_FAILURE
+  stage: RUNNING
+  behavior: PERMANENT
+  reaction: NEVER_RETRY
+  reason: "Container failed due to cude version mismatch"
+  repro:
+  - "Using cude 10 in container with driver version below 410.48"
+  solution:
+  - "Change docker image with correct cuda runtime version"
+  - "Contact Cluster Admin"
+  pattern:
+    runtimeContainerPatterns:
+    - userLogRegex: "cuda driver version is insufficient for cuda runtime version"
+
+- code: 225
+  phrase: ContainerCudaDnnVersionMismatch
+  issuer: PAI_RUNTIME
+  causer: USER_CONTAINER
+  type: USER_FAILURE
+  stage: RUNNING
+  behavior: PERMANENT
+  reaction: NEVER_RETRY
+  reason: "Container failed due to cuda dnn version mismatch"
+  repro:
+  - "Running latest pytorch with old cuDNN lib"
+  solution:
+  - "Change docker image with correct DL framework version"
+  pattern:
+    runtimeContainerPatterns:
+    - userLogRegex: "dropout supported only in cudnn v"
+
+- code: 226
+  phrase: ContainerGpuOOMKilled
+  issuer: PAI_RUNTIME
+  causer: USER_CONTAINER
+  type: USER_FAILURE
+  stage: RUNNING
+  behavior: PERMANENT
+  reaction: NEVER_RETRY
+  reason: "Container failed due to out of gpu memory"
+  repro:
+  - "Allocate GPU memory exceed hardware limit"
+  solution:
+  - "Select GPU type which has more physical memory"
+  - "Tuning the program parameter to use less memory"
+  pattern:
+    runtimeContainerPatterns:
+    - userLogRegex: "cuda runtime error (2) : out of memory"
+
+- code: 227
+  phrase: ContainerEnginCoreDump
+  issuer: PAI_RUNTIME
+  causer: USER_CONTAINER
+  type: USER_FAILURE
+  stage: RUNNING
+  behavior: PERMANENT
+  reaction: NEVER_RETRY
+  reason: "Container failed due to engine core dump"
+  repro:
+  - "Access invalid memory address"
+  solution:
+  - "Correct the user code"
+  pattern:
+    runtimeContainerPatterns:
+    - userLogRegex: "segmentation fault"
+    - userLogRegex: "core dumped"
```

Comment:
Why include this into engine core dump

Reply:
According to log pattern in Philly, use log will print this when core dump occur. But don't find any clue when searching it. Remove it at this time

---
In reply to: [353553979](https://github.com/microsoft/pai/pull/3959#discussion_r353553979) [](ancestors = 353553979)

Reply:
ContainerEngineCoreDump is not segmentation fault

---
In reply to: [353563197](https://github.com/microsoft/pai/pull/3959#discussion_r353563197) [](ancestors = 353563197,353553979)